docs/navigation/activities.md

```diff
@@ -61,31 +73,87 @@ class ShareRoute(
         })
 }
 
-val shareDestination: NavDestination = ActivityDestination<ShareRoute>(
-    // basic intent that is extended with the Intent above
-    intent = Intent(Intent.ACTION_CHOOSER)
-)
-```
-
-Another example which would open a browser with a given `Uri`:
-
-```kotlin
 // route with a data uri extra
 @Parcelize
 class BrowserRoute(
-    uri: Uri,
+    private val uri: Uri,
 ) : ExternalActivityRoute {
     // the returned Intent is filled into the Intent of the destination by calling
     // destinationIntent.fillIn(fillInIntent())
-    override fun fillInIntent() = Intent().setData(uri)
+    override fun fillInIntent() = Intent(Intent.ACTION_VIEW).setData(uri)
 }
+```
+
+## Activity results
 
-val browserDestination: NavDestination = ActivityDestination<BrowserRoute>(
-    // basic intent that is extended with the Intent above
-    intent = Intent(Intent.ACTION_VIEW)
-)
+External SDKs and the framework often provide an `Activity` that is supposed to be used with
+`startActivityForResult`. AndroidX already introduced `ActivityResultContract` to simplify handling
+this and `ActivityNavigator` uses them to also enable starting them from outside the UI layer
```

Comment:
rephrased since it's not obvious what `them` is

```suggestion
this and `ActivityNavigator` uses it to also enable starting Activities from outside the UI layer
```